File: LeetCode/LeetCode Practice Problems/1370- Increasing Decreasing String.cpp
class Solution {
public:
    string sortString(string s) {
        string ans = "";
        int f[26];
        for(int i=0;i<26;i++)
            f[i] = 0;
        
        for(auto i : s)
            f[i - 'a']++;
        
        bool c = true;
        while(c)
        {
            c = false;
            for(int i=0;i<26;i++)
            {
                if(f[i])
                {
                    c = true;
                    ans = ans + (char)(i + 97);    
                    f[i]--;
                }
            }
            
            for(int i=25;i>=0;i--)
            {
                if(f[i])
                {
                    ans = ans + (char)(i + 97);
                    f[i]--;
                }
            }
        }
        return ans;
    }
};

File: LeetCode/LeetCode Practice Problems/1370-Increasing Decreasing String.java
class Solution {
    public String sortString(String s) {
        int cnt[] = new int[26];
        int c = 0;
        
        for(char ch:s.toCharArray()) {
            cnt[ch - 'a']++;
            c++;
        }
        
        boolean f = true;
        String ans = "";
        
        while(c > 0) {
            if(f) {
                for(int i=0;i<26;i++) {
                    if(cnt[i] == 0) continue;
                    ans = ans + (char)(i + 'a');
                    cnt[i]--;
                    c--;
                }
            } else {
                for(int i=25;i>=0;i--) {
                    if(cnt[i] == 0) continue;
                    ans = ans + (char)(i + 'a');
                    cnt[i]--;
                    c--;
                }
            }
            f = !f;
        }
        
        return ans;
    }
}
